File: sk000801/입문/외계어 사전.java
class Solution {
    public int solution(String[] spell, String[] dic) {
        for(int i = 0; i < dic.length; i++) {
            String s = dic[i];
            int count = 0;
            for(int j = 0; j < spell.length; j++) {
                int l1 = s.length();
                s = s.replaceFirst(spell[j], "");
                if(l1-s.length() != 0) count++;
            }
            if(s.equals("") && count == spell.length) return 1;
        }
        return 2;
    }
}

File: kyw0716/level0/외계어 사전.js
const isSame = (arr1, arr2) => {
  arr1 = arr1.sort().join("");
  arr2 = arr2.sort().join("");

  return arr1 === arr2;
};

function solution(spell, dic) {
  let answer = false;

  dic = dic.map((d) => d.split(""));
  dic.forEach((d) => {
    if (isSame(d, spell)) answer = true;
  });

  return answer ? 1 : 2;
}

// https://school.programmers.co.kr/learn/courses/30/lessons/120869
